### PR TITLE
feat: update totalFee calculation

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

As per [this internal discussion](https://cowservices.slack.com/archives/C036G0J90BU/p1705083817684659), the backend updated how the fees are returned by the api.

In this PR the calculation is updated to return the sum of `executedSurplusFee` and `executedFeeAmount`.

# Context

Take as example this order: https://explorer.cow.fi/orders/0xdedb65b51a6f81493f8e6e1d811b56a57297dbb6fb11716d85e646a66a5d041f85cc3c8da85612013acabe9b5d954d578860b3c165a22c43?tab=overview

Currently it shows as `0` fee, even though `executedFeeAmount` is set.

The issue is that [this order has both fields](https://api.cow.fi/mainnet/api/v1/orders/0xdedb65b51a6f81493f8e6e1d811b56a57297dbb6fb11716d85e646a66a5d041f85cc3c8da85612013acabe9b5d954d578860b3c165a22c43) `executedSurplusFee` and `executedFeeAmount` returned, but `executedSurplusFee` is set to 0.
In the current implementation, it takes precedence over `executedFeeAmount`.

https://github.com/cowprotocol/cow-sdk/blob/3abfdcfd1ab223394e43d647de07a88d06f0b30d/src/order-book/transformOrder.ts#L28

This logic was [introduced over 1y ago](https://github.com/cowprotocol/cow-sdk/pull/88/files) and according to [this comment from Nick](https://github.com/cowprotocol/cowswap/pull/1588#discussion_r1037501631):
> tl;dr: the “real” fee is executedSurplusFee ?? executedFeeAmount, and not (executedSurplusFee ?? 0) + executedFeeAmount.

According to Dusan:

 > ... probably caused by this change: https://github.com/cowprotocol/services/pull/2103
We extracted the executed_surplus_fee out of the parent struct which caused it to be always serialized (even for market orders).
Considering we will probably deliver protocol fees (which are expressed through executed_surplus_fee ) before we completely remove market orders (orders with signed fee_amount), and in that case we will have orders that have both executed_fee_amount and executed_surplus_fee different from zero => I would go with summarizing the values for UI.

# Test

Unit tests

### CODE REVIEW

File: src/order-book/transformOrder.test.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  status: OrderStatus.FULFILLED,
+}
+
+describe('transformOrder', () => {
+  describe('addTotalFeeToOrder', () => {
+    test('should use executedFeeAmount when executedSurplusFee is 0', () => {
+      const rawOrder = { ...ORDER, executedFeeAmount: '1', executedSurplusFee: '0' }
+      const transformedOrder = transformOrder(rawOrder)
+
+      expect(transformedOrder.totalFee).toEqual(rawOrder.executedFeeAmount)
```
Comment (anxolin) [inline src/order-book/transformOrder.test.ts:35]: nit: Why asserting the `transformedOrder` totalFee matches the rawOrder `executedFeeAmount`? 
I would find more readable that given the above, the result is '1' (setting the actual value, instead of picking the property name)

Comment (anxolin) [reply src/order-book/transformOrder.test.ts:35]:
```suggestion
      expect(transformedOrder.totalFee).toEqual('1') // 1 + 0
```

File: src/order-book/transformOrder.test.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    })
+
+    test('should use executedSurplusFee when executedFeeAmount is 0', () => {
+      const rawOrder = { ...ORDER, executedFeeAmount: '0', executedSurplusFee: '1' }
+      const transformedOrder = transformOrder(rawOrder)
+
+      expect(transformedOrder.totalFee).toEqual(rawOrder.executedSurplusFee)
```
Comment (anxolin) [inline src/order-book/transformOrder.test.ts:42]: same comment

File: src/order-book/transformOrder.ts [hunk 3]
```diff
@@ -16,16 +16,23 @@ export function transformOrder(order: Order): EnrichedOrder {
...
+ *
  * @param dto The order to add the total fee to.
  * @returns The order with the total fee added.
  */
 function addTotalFeeToOrder(dto: Order): EnrichedOrder {
   const { executedFeeAmount, executedSurplusFee } = dto
-  const totalFee = executedSurplusFee ?? executedFeeAmount
+
+  const _executedFeeAmount = BigInt(executedFeeAmount || '0')
+  const _executedSurplusFee = BigInt(executedSurplusFee || '0')
```
[the remark this answers is not in the document]
Comment (anxolin) [reply src/order-book/transformOrder.ts:33]: this solution looks good.

why the underscore in the consts?

[the remark this answers is not in the document]
Comment (alfetopito) [reply src/order-book/transformOrder.ts:33]: Because those names already exist in the scope.
Could call then `name+BigInt` but I didn't find it a good choice.
Suggestions?

[the remark this answers is not in the document]
Comment (anxolin) [reply src/order-book/transformOrder.ts:33]: okok, not big deal. works for me. Also the BigInt works too

**Review by anxolin — APPROVED** [review]
just some nits

File: src/order-book/transformOrder.test.ts (hunk 1 above)
Comment (alfetopito) [reply src/order-book/transformOrder.test.ts:35]: It's meant to be: it's the unaltered value coming from the original order obj.
I feel like that's a better way to ready it than using an actual value.

But sure, will update it.
